app/src/main/java/com/sdp/movemeet/map/MiniMapFragment.java

```diff
@@ -0,0 +1,85 @@
+package com.sdp.movemeet.map;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import androidx.core.app.ActivityCompat;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.tasks.Task;
+import com.sdp.movemeet.R;
+import com.sdp.movemeet.UploadActivityActivity;
+
+import static com.sdp.movemeet.map.MainMapFragment.REQUEST_CODE;
+
+public class MiniMapFragment extends Fragment implements OnMapReadyCallback, GoogleMap.OnMapClickListener {
+
+    private SupportMapFragment supportMapFragment;
+    private FusedLocationProviderClient fusedLocationProviderClient;
+    private Location currentLocation;
+
+    public static MiniMapFragment newInstance() {
+        return new MiniMapFragment();
+    }
+
+     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.fragment_map, container, false);
+
+        supportMapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.google_map);
+
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(supportMapFragment.getActivity());
+        fetchLastLocation();
+
+        return view;
+    }
+
+    //TODO refactor this ?
```

Comment:
If we can guarantee that all actions requiring permissions happen after a permission check, we might be able to refactor these permission request code blocks into functions and then suppress warnings for permission request requirements, but that seems potentially dangerous

Reply:
I made this comment mostly because I'm using the exact same blocks in two different classes (the two fragments for the map). I thought we could maybe create an abstract class used for the permission checks, to satisfy code climate. I didn't have the time to dive into this matter, but it shouldn't be that long.

Reply:
Oh! A whole class might be something to look into indeed